design/one-pager-template-stacks.md

```diff
@@ -0,0 +1,436 @@
+* **CR (Custom Resource)**
+  : An instance of a Kubernetes type that was defined using a CRD
+* **GVK (Group Version Kind)**
+  : The API Group, Version, and Kind for a type of Kubernetes resource (including CRDs)
+* **TS (Template Stack)**
+  : UNDER THIS PROPOSAL, a Stack that declares dependent resources and includes managed CRDs, but includes templates to be rendered for each CR rather than a `Deployment` controller
+* **TSM (Template Stack Manager)**
+  : UNDER THIS PROPOSAL, a restricted `Deployment` that manages resources for a Template Stack's CRDs
+
+## Introduction
+
+### In Scope
+
+This design introduces the concept and benefits of Template Stacks as well as exploring a proposed Template Stack Manager which may facilitate this experience. Implementation challenges for a the existing Stack Manager and the proposed Template Stack Manager (and approaches to overcome them) are provided.
+
+### Out of Scope
+
+Throughout the design, Go templates are used to illustrate potential usage, but this is an implementation detail that will be the subject of additional design documents.
+
+This design is also not overly concerned with the format of the `Stack` object or the layout of the Stack filesystem. Those topics will be discussed in another one-pager.
+
+Finally, the specific resources that a Template Stack may render is not in scope for this design. There is some thinking on this included in [Concerning the use of Core Types](#Concerning-the-use-of-Core-Types).
+
+## Benefits of Template Stacks
+
+The current design of Stacks allow for a new CRD to be managed by an independent controller. A CR will be handled by the Stack controller in any way that controller sees fit. This includes issuing remote API calls for additional inputs or creating external outputs, accessing local resources, or performing complex computations.
+
+Stacks may also opt to take advantage of the Crossplane Runtime to simplify the life cycle management of resources.
+
+In contrast, this proposal introduces Template Stacks with the limited ability to use their CR and child resources as input to create, update, or delete child Kubernetes resources. Application owners and managers can take advantage of Template Stacks to package complex deployments, while users benefit from a simple install and usage interface. Developers can use Template Stacks as application owners would, to create Kubernetes resource machinery using template engine logic, circumventing the need more complex controllers.
+
+The Template Stack Manager will permit the use of template variables to create and manage a set of resources provided as a set of strings included in the definition of the Stack.
+
+The benefits of Template Stacks are as follows:
+
+* [Simple Variable Substitution](#Simple-Variable-Substitution)
+
+  Stack Template resources can take input from a CR spec or from sibling resources.
+* [No Custom Controller Needed](#No-Custom-Controller-Needed)
```

Comment:
I am using a single list here.  Each bullet has additional text underneath it.  What appears to be a blank line is the extra newline needed to prevent the additional text from each bullet from wrapping on the first line of the bullet.

I've included an example here, but it seems that Github Markdown doesn't need the extra newline.  My IDE's markdown preview did need it.  The rendering doesn't suffer from wrapping on the first line. I'll remove it.

```markdown
* **bulletA**

   a second line of text about bulletA
* **bulletB**
   a line of text I expected would wrap into the first line of bulletB
```

---

* **bulletA**

   a second line of text about bulletA
* **bulletB** 
   a line of text I expected would wrap into the first line of bulletB

---